torch_brain/models/poyo.py

```diff
@@ -132,10 +161,87 @@ def __init__(
+        self.readout = nn.Linear(dim, readout_spec.dim)
 
         self.dim = dim
 
+    def tokenize(self, data: Data) -> Dict:
+        # context window
+        start, end = 0, self.sequence_length
+
+        ### prepare input
+        unit_ids = data.units.id
+        spike_unit_index = data.spikes.unit_index
+        spike_timestamps = data.spikes.timestamps
+
+        # create start and end tokens for each unit
+        (
+            se_token_type_index,
+            se_unit_index,
+            se_timestamps,
+        ) = create_start_end_unit_tokens(unit_ids, start, end)
+
+        # append start and end tokens to the spike sequence
+        spike_token_type_index = np.concatenate(
+            [se_token_type_index, np.zeros_like(spike_unit_index)]
+        )
+        spike_unit_index = np.concatenate([se_unit_index, spike_unit_index])
+        spike_timestamps = np.concatenate([se_timestamps, spike_timestamps])
+
+        # unit_index is relative to the recording, so we want it to map it to
+        # the global unit index
+        local_to_global_map = np.array(self.unit_emb.tokenizer(unit_ids))
+        spike_unit_index = local_to_global_map[spike_unit_index]
+
+        ### prepare latents
+        latent_index, latent_timestamps = create_linspace_latent_tokens(
+            start,
+            end,
+            step=self.latent_step,
+            num_latents_per_step=self.num_latents,
+        )
+
+        output_timestamps, output_values, output_weights, eval_mask = (
+            prepare_for_readout(data, self.readout_spec)
+        )
+
+        # create session index for output
+        output_session_index = self.session_emb.tokenizer(data.session)
+        output_session_index = np.repeat(output_session_index, len(output_timestamps))
+
+        batch = {
+            # input sequence
+            "input_unit_index": pad8(spike_unit_index),
+            "input_timestamps": pad8(spike_timestamps),
+            "input_token_type": pad8(spike_token_type_index),
+            "input_mask": track_mask8(spike_unit_index),
+            # latent sequence
+            "latent_index": latent_index,
+            "latent_timestamps": latent_timestamps,
+            # output sequence
+            "output_session_index": pad8(output_session_index),
+            "output_timestamps": pad8(output_timestamps),
+            "output_mask": track_mask8(output_session_index),
+            # ground truth targets
+            "target_values": pad8(output_values),
+            "target_weights": pad8(output_weights),
+        }
+
+        extra = {
+            "session_id": data.session,
+            "absolute_start": data.absolute_start,
+            "eval_mask": (
+                pad8(eval_mask)
+                if eval_mask is not None
+                else track_mask8(output_session_index)
+            ),
+        }
+
+        return batch, extra
+
+    def get_tokenizer(self):
+        return self.tokenize
+
```

Comment:
I don't see a point of this function yet. What's your philosophy behind not using `model.tokenize` directly as the tokenizer transform?